src/runtime/strapi.ts

```diff
@@ -0,0 +1,222 @@
+      }
+
+      const { response: { data: { message: msg } } }: any = err
+
+      let message
+      if (Array.isArray(msg)) {
+        message = msg[0].messages[0].message
+      } else if (typeof msg === 'object' && msg !== null) {
+        message = msg.message
+      } else {
+        message = msg
+      }
+
+      err.message = message;
+      (err as any).original = (err.response as any).data
+      this.callHook('error', err)
+    })
+  }
+
+  get user (): NuxtStrapiUser {
+    return this.state.user
+  }
+
+  set user (user) {
+    Vue.set(this.state, 'user', user)
+  }
+
+  async register (data: NuxtStrapiRegistrationData): Promise<NuxtStrapiLoginResult> {
+    this.clearToken()
+    const { user, jwt } = await this.$http.$post<any>('/auth/local/register', data)
+    this.setToken(jwt)
+    await this.setUser(user)
+    return { user, jwt }
+  }
+
+  async login (data: NuxtStrapiLoginData): Promise<NuxtStrapiLoginResult> {
+    this.clearToken()
+    const { user, jwt } = await this.$http.$post<any>('/auth/local', data)
+    this.setToken(jwt)
+    await this.setUser(user)
+    return { user, jwt }
+  }
+
+  forgotPassword (data: NuxtStrapiEmailData): Promise<unknown> {
+    this.clearToken()
+    return this.$http.$post('/auth/forgot-password', data)
+  }
+
+  async resetPassword (data: NuxtStrapiResetPasswordData): Promise<NuxtStrapiLoginResult> {
+    this.clearToken()
+    const { user, jwt } = await this.$http.$post<any>('/auth/reset-password', data)
+    this.setToken(jwt)
+    await this.setUser(user)
+    return { user, jwt }
+  }
+
+  sendEmailConfirmation (data: NuxtStrapiEmailData): Promise<unknown> {
+    return this.$http.$post('/auth/send-email-confirmation', data)
+  }
+
+  async logout (): Promise<void> {
+    await this.setUser(null)
+    this.clearToken()
+  }
+
+  async fetchUser (): Promise<NuxtStrapiUser> {
+    const jwt = this.syncToken()
+    if (!jwt) {
+      return null
+    }
+
+    try {
+      const user = await this.findOne('users', 'me')
+      await this.setUser(user)
+    } catch (e) {
+      this.clearToken()
+    }
+
+    return this.user
+  }
+
+  async setUser (user): Promise<void> {
+    this.user = user
+    await this.callHook('userUpdated', user)
+  }
+
+  find<T = any, E = string> (entity: E, searchParams?: NuxtStrapiQueryParams): Promise<T> {
+    return this.$http.$get<T>(`/${entity}`, { searchParams })
+  }
+
+  count<T = any, E = string> (entity: E, searchParams?: NuxtStrapiQueryParams): Promise<T> {
+    return this.$http.$get<T>(`/${entity}/count`, { searchParams })
+  }
+
+  findOne<T = any, E = string> (entity: E, id: string): Promise<T> {
+    return this.$http.$get<T>(`/${entity}/${id}`)
+  }
+
+  create<T = any, E = string> (entity: E, data: NuxtStrapiQueryParams): Promise<T> {
+    return this.$http.$post<T>(`/${entity}`, data)
+  }
+
+  // TODO: define `update()` types
+  update (entity, id, data) {
+    if (typeof id === 'object') {
+      data = id
+      id = undefined
+    }
```

Comment:
```suggestion
  update<T = any, E = string> (entity: E, id: string, data: NuxtStrapiQueryParams): Promise<T> {
    if (typeof id === 'object') {
      data = id
      id = undefined
    }
```
There was still this part left to do, I suggest these typings.